std/internal/test/randomized_unittest_benchmark.d

```diff
@@ -0,0 +1,495 @@
+                q100 > this.avgStopWatch ? q100 - this.avgStopWatch : 0
+            );
+        }
+    }
+}
+
+struct GenNum(T,T low, T high)
+{
+    alias Value = T;
+
+    T value;
+
+    void gen(ref Random gen)
+    {
+        this.value = uniform!("[]")(low, high, gen);
+    }
+
+    ref T opCall()
+    {
+        return this.value;
+    }
+
+    alias opCall this;
+}
+
+struct GenString(T, size_t low, size_t high)
+        if(isSomeString!T)
+{
+    alias Value = T;
+
+    T value;
+
+    void gen(ref Random gen)
+    {
+        auto app = appender!T();
+        app.reserve(high);
+        size_t numElems = uniform!("[]")(low, high, gen);
+        for(size_t i = 0; i < numElems; ++i)
+        {
+            app.put(uniform!("[]")(cast(dchar)0x0023, cast(dchar)0x007E, gen));
+        }
+
+        this.value = app.data;
+    }
+
+    ref T opCall()
+    {
+        return this.value;
+    }
+
+    alias opCall this;
+}
+
+unittest
+{
+    import std.ascii;
+
+    auto rnd = Random(1337);
+
+    foreach (S; TypeTuple!(string, wstring, dstring))
+    {
+        GenString!(S, 5, 5) gen;
+        gen.gen(rnd);
+        S str = gen();
+
+        assert(str.length == 5);
+        foreach(it; str)
+        {
+            assert(isASCII(it));
+        }
+    }
+}
+
+private pure T unicodeTable(T)() @safe {
+    import std.range : chain, iota;
+    import std.algorithm : map, joiner;
+    import std.array : array;
+    import std.conv : to;
+
+    auto arr = chain(
+        iota(0x21, 0x7E).map!(a => to!T(cast(dchar)a)),
+        iota(0xA1, 0x1EF).map!(a => to!T(cast(dchar)a)));
+
+    return to!T(arr.joiner.array);
+}
+
+unittest
+{
+    import std.uni : unicode;
+    import std.conv : to;
+    import std.format : format;
+
+    auto charSet = unicode.block("Basic Latin") |
+        unicode.block("Latin-1 Supplement") |
+        unicode.block("Latin Extended-A") |
+        unicode.block("Latin Extended Additional") |
+        unicode.block("Latin Extended-B") |
+        unicode.block("Latin Extended-C") |
+        unicode.block("Latin Extended-D");
+
+    foreach (S; TypeTuple!(string, wstring, dstring))
+    {
+        enum table = unicodeTable!S();
+        foreach(dchar it; table)
+        {
+            assert(charSet[it], format("%c %x", it, cast(int)it));
+        }
+    }
+}
+
+struct GenUnicodeString(T, size_t low, size_t high)
+        if(isSomeString!T)
+{
+    alias Value = T;
+
+    dstring possibleChars = unicodeTable!dstring();
+
+    T value;
+
+    void gen(ref Random gen)
+    {
+        import std.conv : to;
+        auto app = appender!T();
+        app.reserve(high);
+        size_t numElems = uniform!("[]")(low, high, gen);
+        for(size_t i = 0; i < numElems; ++i)
+        {
+            app.put(randomSample(possibleChars, to!size_t(1u)));
+        }
+
+        this.value = app.data;
+    }
+
+    ref T opCall()
+    {
+        return this.value;
+    }
+
+    alias opCall this;
+}
```

Comment:
This seems inefficient to me: generate full Unicode tables and then select elements from them. It seems an awful lot simpler and cheaper to just generate one correct random Unicode character at a time.